network-store-model/src/main/java/com/powsybl/network/store/model/Resource.java

```diff
@@ -55,9 +59,14 @@ public String getMessageHeader() {
         return type.getDescription() + " '" + id + "': ";
     }
 
-    public static <T extends IdentifiableAttributes> Resource<T> create(ResourceType type, String id, int variantNum, T attributes) {
+    public Resource<Attributes> filterAttributes(AttributeFilter filter) {
+        Objects.requireNonNull(filter);
+        return new Resource<>(type, id, variantNum, filter, ((IdentifiableAttributes) attributes).filter(filter));
```

Comment:
maybe we should refactor to pass the attributes in a separate argument as the resource, so that we don't have to clone the resource here ?

Reply:
I tried but I don't think it is possible to do that in an elegant and maintainable way.
The cloning has a very low impact on performance.